File: examples/dom-mock/dom.js
const { JSDOM } = require("jsdom")

const dom = new JSDOM(
  `
  <!DOCTYPE html>
  <html>
    <head>
      <title>Go wasm</title>
    </head>
    <body></body>
  </html>
  `,
  {
    url: "http://localhost:8080/wasm_exec.html"
  },
)

global.window = dom.window
global.document = window.document

File: examples/dom-mock/dom.go
// +build js,wasm

package dom

import (
	"syscall/js"
)

func GetWindowLocation() string {
	return js.Global().Get("window").Get("location").String()
}

func GetDocumentTitle() string {
	return js.Global().Get("document").Get("title").String()
}
